library.rb

```diff
@@ -0,0 +1,218 @@
+    elsif @name.empty?
+      raise ValidationError, "name must be not empty"
+    end
+    if !@biography.is_a?(String)
+      raise ValidationError, "name must be string"
+    elsif @biography.nil?
+      raise ValidationError, "name must be given"
+    end
+  end
+end
+
+class Order
+  attr_reader :book, :reader, :date
+
+  def initialize(book, reader, date = Date.today)
+    @book = book
+    @reader = reader
+    @date = date
+    validate_order
+  end
+
+  def to_s
+    "{ book: #{book}, reader: #{reader}, date: #{date} }"
+  end
+
+  private
+
+  def validate_order
+    if @book.nil?
+      raise ValidationError, "name must be given"
+    elsif @book.is_a?(Book)
+      raise ValidationError, "name must be string"
+    end
+    if @reader.nil?
+      raise ValidationError, "name must be given"
+    elsif @reader.is_a?(Reader)
+      raise ValidationError, "name must be string"
+    end
+  end
+end
+
+class Reader
+  attr_reader :name, :email, :city, :street, :house
+
+  def initialize(name, email, city, street, house)
+    @name = name
+    @email = email
+    @city = city
+    @street = street
+    @house = house
+    validate_reader
+  end
+
+  def to_s
+    "{ name: #{name}, email: #{email}, city: #{city}, street: #{street}, house: #{house} }"
+  end
+
+  private
+
+  def validate_reader
+    if @name.nil?
+      raise ValidationError, "name must be given"
+    elsif !@name.is_a?(String)
+      raise ValidationError, "name must be string"
+    elsif @name.empty?
+      raise ValidationError, "name must be not empty"
+    end
+    if @email.nil?
+      raise ValidationError, "name must be given"
+    elsif !@email.is_a?(String)
+      raise ValidationError, "name must be string"
+    elsif @email.empty?
+      raise ValidationError, "name must be not empty"
+    end
+    if @city.nil?
+      raise ValidationError, "name must be given"
+    elsif !@city.is_a?(String)
+      raise ValidationError, "name must be string"
+    elsif @city.empty?
+      raise ValidationError, "name must be not empty"
+    end
+    if @street.nil?
+      raise ValidationError, "name must be given"
+    elsif !@street.is_a?(String)
+      raise ValidationError, "name must be string"
+    elsif @street.empty?
+      raise ValidationError, "name must be not empty"
+    end
+    if @house.nil?
+      raise ValidationError, "name must be given"
+    elsif !@house.is_a?(Integer)
+      raise ValidationError, "name must be string"
+    elsif !@house.positive?
+      raise ValidationError, "name must be not empty"
+    end
+  end
+end
+
+library = Library.new
+author = Author.new('0', '1')
+book = Book.new('1','1')
+reader = Reader.new('name1', 'email1', 'city1', 'street1', 1)
+order = Order.new('0', 0)
+library.add_book(book)
+library.add_author(author)
+library.add_order(order)
+library.add_reader(reader)
+# author_error_nil = Author.new(0, '2')
+# author_error_class = Author.new()
+# author_error_empty = Author.new()
+#<class_name_downcase>_<atrribute>_<validation>
```

Comment:
Add examples with `Book`, `Reader`, `Author`, `Order` creation for cases when validation error will be raised e.g.

if validation error raised for cases when name is not provided, then you should call varialble like `author_name_nil`
```ruby
author_name_nil = Author.new(nil, 'biography')
#OR
author_biography_class = Author.new('Name', Array.new)
```

All this examples should be commented out, to make testing easier.

Reply:
```ruby
begin
  book_title_nil = Book.new(nil, Author.new(...))
rescue ValidationError => error
  puts "Test validation for book title is nil: #{error.message}"
end
```